app/src/main/java/org/schabi/newpipe/local/playlist/LocalPlaylistFragment.java

```diff
@@ -331,6 +358,121 @@ public void onComplete() { }
         };
     }
 
+    @Override
+    public boolean onOptionsItemSelected(final MenuItem item) {
+        switch (item.getItemId()) {
+            case R.id.menu_item_removeWatched:
+                android.app.AlertDialog.Builder builder =
+                new android.app.AlertDialog.Builder(getActivity());
+                builder.setMessage(R.string.remove_watched_popup_warning)
+                        .setTitle(R.string.remove_watched_popup_title)
+                        .setPositiveButton(R.string.remove_watched_popup_yes,
+                                (DialogInterface d, int id) -> removeWatchedStreams(false))
+                        .setNeutralButton(
+                                R.string.remove_watched_popup_yes_and_partially_watched_videos,
+                                (DialogInterface d, int id) -> removeWatchedStreams(true))
+                        .setNegativeButton(R.string.remove_watched_popup_cancel,
+                                (DialogInterface d, int id) -> d.cancel());
+                builder.create().show();
+                break;
+            default:
+                return super.onOptionsItemSelected(item);
+        }
+        return true;
+    }
+
+    public void removeWatchedStreams(final boolean removePartiallyWatched) {
+        if (removeWatchedDisposable != null && !removeWatchedDisposable.isDisposed()) {
+            // already running
+            return;
+        }
+        showLoading();
+
+        removeWatchedDisposable = Flowable.just(Flowable.just(removePartiallyWatched,
```

Comment:
Do you really need to pass `removePartiallyWatched` to the `Flowable`? I think you can access it just as normal inside the lambda body below

Reply:
I'm really new with ReactiveX, I did that just to make sure that there would be no chance of a weird issue occurring, do you think I should change it?

Reply:
Yes, nothing strange should happen, as `removePartiallyWatched` is declared as final

Reply:
I tried using that just now, and it causes the removeWatchedDisposable to never dispose.
So you can only use the 'remove watched' ability once, then you have to shut down and reopen the app.

I found a way around it, by force disposing it.

Reply:
Fixed in db335d5